velox/exec/Spiller.cpp

```diff
@@ -203,14 +207,26 @@ std::unique_ptr<SpillerBase::SpillStatus> SpillerBase::writeSpill(
   constexpr int32_t kTargetBatchRows = 64;
 
   RowVectorPtr spillVector;
+  SpillRows spillRows(*memory::spillMemoryPool());
   auto& run = spillRuns_.at(id);
   try {
     ensureSorted(run);
     size_t written = 0;
     while (written < run.rows.size()) {
-      extractSpillVector(
-          run.rows, kTargetBatchRows, kTargetBatchBytes, spillVector, written);
-      state_.appendToPartition(id, spillVector);
+      if (serializeRowContainer_) {
+        extractSpillRows(
+            run.rows, kTargetBatchRows, kTargetBatchBytes, spillRows, written);
+        state_.appendToPartition(id, spillRows, container_);
+        spillRows.clear();
```

Comment:
Do not need to call clear, this is the last call of the function

Reply:
Maybe move SpillRows spillRows(*memory::spillMemoryPool()); to after if (serializeRowContainer_) {